File: BookShop.BusinessLogic/Models/PrintedEditionModels/PrintedEditionModel.cs
using BookShop.BusinessLogic.AuthorModels;
using BookShop.DataAccess.Entities;
using BookShop.DataAccess.ObjectModels.PrintedEditionWithNestedObjects;
using System.Collections.Generic;
using System.Linq;
using static BookShop.DataAccess.Common.Enums.EntityFields;

namespace BookShop.BusinessLogic.PrintedEditionModels
{
    public class PrintedEditionModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public bool IsRemoved { get; set; }
        public Currency Currency { get; set; }
        public PrintedEditionType Type { get; set; }
        public List<AuthorModel> AuthorModels { get; set; }
        public PrintedEditionModel()
        {
        }
        internal PrintedEditionModel(PrintedEdition entity) : this()
        {
            Id = entity.Id;
            Name = entity.Name;
            Description = entity.Description;
            Price = entity.Price;
            IsRemoved = entity.IsRemoved;
            Currency = entity.Currency;
            Type = entity.Type;
        }

        internal PrintedEditionModel(PrintedEditionWithNestedObjects printedEdition) : this(printedEdition.PrintedEdition)
        {
            if (!(printedEdition.AuthorInBooks is null))
            {
                AuthorModels = printedEdition.AuthorInBooks.Select(item => new AuthorModel(item.Author)).ToList();
            }
        }

        internal PrintedEdition MapToEntity(PrintedEdition entity = null)
        {
            if (entity is null)
            {
                entity = new PrintedEdition();
            }
            entity.Name = Name;
            entity.Description = Description;
            entity.Price = Price;
            entity.IsRemoved = IsRemoved;
            entity.Currency = Currency;
            entity.Type = Type;
            return entity;
        }
    }
}

File: BookShop.Presentation/wwwroot/ClientApp/src/app/models/printedEditionModels/printedEditionModel.ts
import { PrintedEditionType } from 'src/app/enums/printedEditionType';
import { Currency } from 'src/app/enums/currency';
import { AuthorModel } from 'src/app/models/authorModels/authorModel';

export class PrintedEditionModel {
    public id: number;
    public name: string;
    public description: string;
    public price: number;
    public isRemoved: boolean;
    public type: PrintedEditionType;
    public currency: Currency;
    public authorModels: AuthorModel[];

    constructor() {
        this.initializeDefaultModel();
    }

    private initializeDefaultModel() {
        this.type = PrintedEditionType.None;
        this.currency = Currency.None;
        this.authorModels = [];
        this.price = 0;
    }
}
